src/Data/Serialize/Get.hs

```diff
@@ -402,30 +418,93 @@ ensure n0 = n0 `seq` Get $ \ s0 b0 m0 w0 kf ks -> let
                     in ks s b m0 w0 s
                 else getMore n' s0 ss b0 m0 w0 kf ks
 
+negativeIsolation :: Get a
+negativeIsolation = fail "Attempted to isolate a negative number of bytes"
+
+isolationUnderParse :: Get a
+isolationUnderParse =  fail "not all bytes parsed in isolate"
+
+isolationUnderSupply :: Get a
+isolationUnderSupply = failRaw "too few bytes" ["demandInput"]
```

Comment:
Why use `failRaw` here, but not in `isolationUnderParse`?

Reply:
I think I was trying to get the errors throws by `isolateLazy` to match those thrown by `isolate`, so that I could use quickcheck to compare them against each other.

I guess that now that I've done that, I could tweak the errors a bit, make them more informative.